File: Data-structures/linkedlist.java
import java.util.*;
import java.io.*;

class Node{
	int data;
	Node next;

	Node(int data){
		this.data = data;
	}
}

public class linkedlist{
	Node head;

	// function to add new node at the end
	public void append(int data){
		if(head==null){
			head = new Node(data);
			return;
		}
		Node current = head;
		while(current.next!=null){
			current = current.next;
		}
		current.next = new Node(data);
	}

	// function to add new node at the beginning
	public void prepand(int data){
		Node newNode = new Node(data);

		if(head == null){
			head = newNode;
		}
		newNode.next = head;
		head = newNode;
	}

	// function to display all the nodes in Linkedlist
	public void display(){
		Node t = head;

		if(head ==null){
			System.out.println("Linked list is empty.");
			return;
		}

		while(t!=null){
			System.out.println(t.data);
			t = t.next;
		}
	}

	// function used to find the number of nodes in  linkedlist
	public void countNode(){
		if(head==null){
			System.out.println("Linked list is empty");
			return;
		}

		int count=0;
		Node t = head;
		while(t!=null){
			count++;
			t = t.next;
		}
		System.out.println("Number of nodes in Linked list: "+count);
	}

	public static void main(String []args){

		linkedlist ll = new linkedlist();
		ll.append(10);
		ll.append(12);
		ll.append(14);
		ll.append(15);
		
		ll.prepand(5);

		ll.countNode();

		ll.display();
	}
}

File: Data-structures/linkedlist.py
class Node:
	def __init__(self,data):
		self.data = data
		self.next = None

class LinkedList(object):
	def __init__(self):
		self.head = None

	def append(self,data):
		"""
		function to add new node at the end
		"""
		if self.head==None:
			self.head = Node(data)
		else:
			current = self.head
			while current.next != None:
				current = current.next
			current.next = Node(data)


	def prepend(self,data):
		"""
		function to add new node at the beginning
		"""
		newNode = Node(data)
		if self.head == None:
			self.head = newNode
		else:
			newNode.next = self.head
			self.head = newNode


	def display(self):
		"""
		function to display all the nodes in Linkedlist
		"""
		t = self.head
		if self.head == None:
			print("linked list is empty")
			return -1

		while t != None:
			print(t.data)
			t = t.next

	def countNode(self):
		"""
		function to find the number of nodes in the linked list
		"""
		if self.head == None:
			print("Linked list is empty")
		else:
			count = 0
			t = self.head
			while t.next != None:
				count += 1
				t = t.next
			print("Total number of nodes in the linked list are: ",count)

if __name__ == "__main__":

	ll = LinkedList()
	ll.append(10);
	ll.append(12);
	ll.append(14);
	ll.append(15);
		
	ll.prepend(5);

	ll.countNode();

	ll.display();